rest/model/task_annotations.go

```diff
@@ -0,0 +1,128 @@
+package model
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/task_annotations"
+)
+
+type APIAnnotationSource struct {
+	Author *string    `json:"author"`
+	Time   *time.Time `json:"time"`
+}
+type APIIssueLink struct {
+	URL      *string `json:"url"`
+	IssueKey *string `json:"issue_key"`
+}
+type APITaskAnnotation struct {
+	Id            *string             `json:"id"`
+	Note          *string             `json:"note"`
+	Issues        []APIIssueLink      `json:"issues"`
+	SuspectIssues []APIIssueLink      `json:"suspected_issues"`
+	Source        APIAnnotationSource `json:"source"`
+	Metadata      *string             `json:"metadata"`
+}
+
+// APIAnnotationSourceBuildFromService takes the task_annotations.AnnotationSource DB struct and
+// returns the REST struct *APIAnnotationSource with the corresponding fields populated
+func APIAnnotationSourceBuildFromService(t task_annotations.AnnotationSource) *APIAnnotationSource {
+	m := APIAnnotationSource{}
+	m.Author = StringStringPtr(t.Author)
+	m.Time = TimeTimeTimeTimePtr(t.Time)
```

Comment:
this must have been codegen 😄 
(timetimetimetime...)

Reply:
yes :)